### ISSUE TITLE
Git GUI pinned to taskbar appears as "Wish Application"

### ISSUE THREAD

**Issue by daveheq** [git-for-windows/git#737]
I just installed Git for Windows (defaults) and when I pin Git GUI to the taskbar, closing it shows a genie lamp named "Wish Application", and when I click the icon to open Git GUI, it opens some empty Wish box and a box under it with command line (not like Git Bash, it's like white Notepad).

What is going on? I've never had the Wish Application, I was using Git Extensions and msysgit and Git GUI was fine, but Git for Windows uninstalled/dereferenced it and I'm not sure how to fix this.

git version 2.8.1.windows.1

**Comment by shiftkey** [comment]
@daveheq screenshot?

**Comment by daveheq** [comment]
![image](https://cloud.githubusercontent.com/assets/11671656/14657431/1a62be40-064b-11e6-985d-d1fef5d5ca04.png)

**Comment by shiftkey** [comment]
I've never heard of this `Wish Application`, and even Google isn't helping me understand it, so I'm curious and a bit concerned about where this is coming from.

A couple of questions:

 - Did you use the official installer from the [Releases](https://github.com/git-for-windows/git/releases) page? If not, which source?
 - Which OS are you running?
 - If you open PowerShell and run this command does it show you anything interesting?

`Get-WmiObject -Class Win32_Product | Where-Object { $_.Name -like "*Wish*" }`

**Comment by daveheq** [comment]
- I downloaded from here (Google search): https://git-scm.com/download/win
- I'm using Windows 7 Pro 64-bit
- I don't have PowerShell

I uninstalled Git for Windows and downloaded from the Releases page

**Comment by daveheq** [comment]
I just installed it from the Releases page and it's doing the same thing.

**Comment by shiftkey** [comment]
Was able to reproduce this on Win7 64-bit with `Git-2.8.1-64-bit.exe`. Seems to be isolated to the Git GUI executable:

<img width="435" src="https://cloud.githubusercontent.com/assets/359239/14658265/d286a84e-06d5-11e6-8342-0d5ed9dadb82.png">

Nothing evident in the file's properties about this "Wish Application":

<img width="430" src="https://cloud.githubusercontent.com/assets/359239/14658290/04c3c8aa-06d6-11e6-9168-a8e3d5e18ab1.png">

Probably something related to packaging `git-gui`. I'll let @dscho chime in here, but I don't think it's as serious as first thought.

@daveheq do you know what version of Git were you running before this?

**Comment by daveheq** [comment]
I had Git-1.9.4-preview20140929.exe installed previously. Git GUI would behave just like any other Windows app. The problem now is I can't open Git GUI by clicking the taskbar icon, it just opens these weird feather windows. 
![image](https://cloud.githubusercontent.com/assets/11671656/14658792/c84d0cce-0653-11e6-94e2-55012a75eb56.png)

**Comment by daveheq** [comment]
This issue tracker is awesome by the way.

**Comment by daveheq** [comment]
Should I use Git-1.9.5 or an earlier version of Git-2.x?

**Comment by shiftkey** [comment]
Yep, I see something similar when launching the pinned app:

<img width="533" alt="screen shot 2016-04-20 at 9 29 18 am" src="https://cloud.githubusercontent.com/assets/359239/14658856/699e0c50-06da-11e6-82dd-485329a43d77.png">

As a workaround, using a shortcut on the desktop for Git GUI works for me. You can actually pin this shortcut to the desktop, but this then shows the actual problem at play:

<img width="520" alt="screen shot 2016-04-20 at 9 34 42 am" src="https://cloud.githubusercontent.com/assets/359239/14659073/169a5e30-06dc-11e6-9eec-bbc0c7649081.png">

Opening up procmon you get a better idea of what's happening:

<img width="661" src="https://cloud.githubusercontent.com/assets/359239/14659061/fe93b44e-06db-11e6-96fa-a254cde613e8.png">

The first section is `git-gui` launching `wish.exe` which then runs some `git` commands. That seems to be the real app. The second section, I'm not so sure. This is the point where my knowledge stops, so I wish I was more help here.

> Should I use Git-1.9.5 or an earlier version of Git-2.x?

I don't know how far back this goes, and given it's the first report of this I'm not sure on the urgency of this, and the work required to get it integrated with the taskbar correctly may be non-trivial. If you can live without pinning Git GUI to the taskbar then I'd suggest sticking with Git 2.x - there's lots of features and performance gains in Git 2.x that you're missing out on by reverting...

**Comment by shiftkey** [comment]
Side note: this isn't tied to a specific OS - same behaviour on Win10:

<img width="299" src="https://cloud.githubusercontent.com/assets/359239/14659258/95e95a64-06dd-11e6-9b97-9c2d11edf72c.png">

**Comment by shiftkey** [comment]
Updated the title to reflect what we've discovered.

**Comment by dscho** [comment]
> I've never heard of this Wish Application

Git GUI is a Tcl/Tk script, and the shell to execute such scripts is called `wish` (for "WIndowing SHell", I guess).

To pin it correctly to the taskbar, I think we'll have to figure out a way how to set the [`RelaunchCommand` property](https://msdn.microsoft.com/en-us/library/windows/desktop/dd391571%28v=vs.85%29.aspx) from Tcl/Tk.

**Comment by daveheq** [comment]
Pinning the Git GUI shortcut only pins the shortcut; clicking it opens another box at the end of the taskbar of the actual Git GUI interface, which is a bit out of the way and I group my related pinned apps together that I use often. So either I have to deal with a mostly-useless slot in the taskbar with the box I want at the end, or not have a pinned shortcut at all. Either way is inconvenient and adds time, and is the only app I'm having a problem with.

I could pin the previous Git GUI to the taskbar just fine without extraneous boxes and it would open the proper program in the right place, so I don't know why it's not possible now.

I had a similar problem with Eclipse and Netbeans, and I read I could just change a config file setting, but it worked (it would open the splash screen box in the taskbar at the end then close when the app would open).

**Comment by daveheq** [comment]
I found it was easier to use with the shortcut pinned to Start menu than taskbar.

**Comment by dscho** [comment]
> Pinning the Git GUI shortcut only pins the shortcut

@daveheq yes, yes, I know, and I already provided the clue how to fix it: the `RelaunchCommand` property.

> I could pin the previous Git GUI to the taskbar just fine without extraneous boxes and it would open the proper program in the right place, so I don't know why it's not possible now.

Yes, you don't know ;-)

`Git GUI` used to be backed by a `.cmd` file but now is an `.exe`. When spawning programs from a `.cmd`, the *Pin to Taskbar* function remembers the original command-line, not so when spawning from `.exe` files.

**Comment by JohnEffland** [comment]
Here's a few more details on wish.exe from sysinternals running on Win 7 64 bit (showing image from git installed from Git-2.8.1-64-bit.exe):

![image](https://cloud.githubusercontent.com/assets/3935018/14708905/cb3f2e96-079b-11e6-9553-be72cba7232a.png)

**Comment by daveheq** [comment]
@dscho: As far as the "clue" you provided to fix it, you said "I think we'll have to figure out a way how to set the RelaunchCommand property from Tcl/Tk", so does "we" include me? I don't have the foggiest clue how to do that, and the link you left might as well be Greek to me (I don't have time to figure it out anyways unless there's just some instructions there I'm not recognizing), so I'm not even sure if that was intended for me.

As far as the taskbar spawning programs from .exe vs .cmd, every other program I have pinned came from an .exe file as far as I can tell (I either pinned them from the .exe file, its shortcut, or from the taskbar while the program was running). I'm not seeing that any of the programs I pin come from .cmd files, and I've seen in the past (like with Chrome) that shortcuts with command-line options can be pinned and behave as expected.

Let me know if you need any other details.

**Comment by dscho** [comment]
> does "we" include me?

@daveheq I thought you were interested in seeing this issue resolved, so yes, I thought of "we" as "we", not as "I".

**Comment by daveheq** [comment]
Yes, I am interested in seeing it resolved, but when you say "we'll have to
figure out a way how to set the RelaunchCommand property from Tcl/Tk", I
don't think that's something I can help figure out (I'm not educated in
Windows programming and I don't have time to figure it out right now). I
just wanted to report the issue.

On Fri, Apr 22, 2016 at 12:59 AM, Johannes Schindelin <
notifications@github.com> wrote:

> does "we" include me?
>
> @daveheq <https://github.com/daveheq> I thought you were interested in
> seeing this issue resolved, so yes, I thought of "we" as "we", not as "I".
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/git-for-windows/git/issues/737#issuecomment-213293087>
>



-- 
Sent from my work computer

**Comment by dscho** [comment]
> Yes, I am interested in seeing it resolved

Good.

> I don't think that's something I can help figure out

Same here, for different reasons :smiley:

**Comment by daveheq** [comment]
@dscho I'm just trying to report a problem and help any way I can get it resolved, and clarify any ambiguous statements, such as your "we" statement with "clue" and "figure it out" (helping figuring it out is not the same as being interested in seeing it resolved). I hope the information here is clarified.

**Comment by dscho** [comment]
@daveheq sadly, I do not have any time left to work on this. Hopefully somebody will pick up where we (as in "you and me") left things.

**Comment by daveheq** [comment]
You don't have time, you make time.

**Comment by PhilipOakley** [comment]
> You don't have time, you make time.

Is that what you meant to say? (was it rhetorical, or quizical, etc..)

I'm constantly amazed at what @dscho does achieve from his 24/7 level of contribution. Time is one thing that you can't make more of, especially if it's all allocated.

**Comment by daveheq** [comment]
It wasn't a question, it was a statement; people are constantly saying "I don't have time" (though dscho didn't say that exactly, he said he doesn't have any time left, which can mean ever), and it really comes down to them making time, so people are really using the inappropriate word. Now I wonder if dscho is just taking issue with me clearing up what he meant by "we" and "figure it out", and his disparaging comments on my ability to help with this project even if I coded in the language were neither useful nor called for.

I've upgraded to 1.9.5 by the way, which is problem-free. Something like getting the taskbar pin to work like other programs doesn't seem like a difficult thing to do, and I'm surprised it's the only program I've worked with with this "wish" problem; it's very odd, I think something was simply overlooked.

**Comment by dscho** [comment]
It's clear now that nobody will do anything about this issue, everybody is content to talk. So I close this issue.

**Comment by daveheq** [comment]
This is still an issue dscho; it's clear you don't want to fix it because you were offended by the talk, but OK, that's on you.

**Comment by dscho** [comment]
@daveheq I do not think that it is even possible for you to offend me.

What happened is this: you reported an issue, I had to estimate how important you think it is to resolve it. For such an estimate I typically take user's feedback and willingness to work at least *a little bit* on the resolution into account.

It is quite obvious that you do not deem this important enough to tell *yourself* to "make some time" to learn the things you need to learn to address this problem. In contrast, you seemed to be only happy to tell others what you think they should do. Yes, it took a bit more of my time than I would have liked to, but I also got several valuable conclusions out of this ticket.

In short: your actions totally agree with my assessment that this bug is not worth the effort to fix it.

**Comment by daveheq** [comment]
I'd love to work on it but that requires me learning the language which would take far more time than someone who already knows it to fix it, and that says nothing about my capability to fix it or learn the language well. I saw no reason for the issue to be closed, as it can be left open (even as the lowest priority) until someone can work on it, as is typical in these kinds of trackers. Closing it indicates you simply don't want the issue (or at least this specific issue number) revisited, so it isn't treating the issue fairly. Hopefully if someone else opens a new issue on it, it won't be closed and will simply be prioritized properly and left on the list.

**Comment by dscho** [comment]
> I saw no reason for the issue to be closed, as it can be left open

If you want it open, you work on it. If you do not work on it, nobody will.

**Comment by daveheq** [comment]
Nobody will? You're predicting what everyone will do here? OK. Out.

**Comment by dscho** [comment]
> Nobody will? You're predicting what everyone will do here? OK. Out.

It was a figure of speech to reflect what my experience tells me. And your reaction proved me 100% correct :stuck_out_tongue_winking_eye:
